src/main/resources/com/google/api/codegen/js/sample.snip

```diff
@@ -0,0 +1,206 @@
+              console.log({@class.pageStreaming.pageVarName});
+            @end
+
+            if ({@class.responseVarName}.result.nextPageToken) {
+              @if class.pageStreaming.isResourceSetterInRequestBody
+                {@class.requestBodyVarName}.pageToken = {@class.responseVarName()}.result.nextPageToken;
+              @else
+                {@class.paramsVarName}.pageToken = {@class.responseVarName()}.result.nextPageToken;
+              @end
+              {@class.pageStreaming.executeRequestFuncName}({@argList(class, @FALSE)});
+            }
+          }, function(reason) {
+            console.log('error: ' + reason.result.error.message);
+          });
+        }
+
+      @end
+      function {@class.initClientFuncName}() {
+        gapi.client.init({
+          'apiKey': '',  // TODO: Update placeholder with desired API key.
+          @if class.auth.type != "API_KEY"
+            'clientId': '',  // TODO: Update placeholder with desired client ID.
+            'scope': '{@scopeString(class.auth.scopes)}',
+          @end
+          'discoveryDocs': ['https://content.googleapis.com/discovery/v1/apis/{@class.apiName}/{@class.apiVersion}/rest'],
+        }).then(function() {
+          @if class.auth.type != "API_KEY"
+            gapi.auth2.getAuthInstance().isSignedIn.listen({@class.updateSignInStatusFuncName});
+            {@class.updateSignInStatusFuncName}(gapi.auth2.getAuthInstance().isSignedIn.get());
+          @else
+            {@class.makeApiCallFuncName}();
+          @end
+        });
+      }
+
+      function {@class.handleClientLoadFuncName}() {
+        @if class.auth.type != "API_KEY"
+          gapi.load('client:auth2', {@class.initClientFuncName});
+        @else
+          gapi.load('client', {@class.initClientFuncName});
+        @end
+      }
+      @if class.auth.type != "API_KEY"
+
+        function {@class.updateSignInStatusFuncName}(isSignedIn) {
+          if (isSignedIn) {
+            {@class.makeApiCallFuncName}();
+          }
+        }
+
+        function {@class.handleSignInClickFuncName}(event) {
+          gapi.auth2.getAuthInstance().signIn();
+        }
+
+        function {@class.handleSignOutClickFuncName}(event) {
+          gapi.auth2.getAuthInstance().signOut();
+        }
+      @end
+      </script>
+      <script async defer src="https://apis.google.com/js/api.js"
+        onload="this.onload=function(){};{@class.handleClientLoadFuncName}()"
+        onreadystatechange="if (this.readyState === 'complete') this.onload()">
+      </script>
+      @if class.auth.type != "API_KEY"
+        <button id="signin-button" onclick="{@class.handleSignInClickFuncName}()">Sign in</button>
+        <button id="signout-button" onclick="{@class.handleSignOutClickFuncName}()">Sign out</button>
+      @end
+    </body>
+  </html>
+@end
+
+# If `isApiCall` is true, then if `needParams` is false and `hasRequestBody` is
```

Comment:
I'd word this more clearly as:
```
If `needParams` is false and `hasRequestBody` is true, then the function will return "{}, {@class.requestBodyVarName}" if `isCall` is true, otherwise will return "{@class.requestBodyVarName}". The first case is needed for calls to the function dynamically generated by the client library, which takes the parameter list as first argument. The second case is used to simplify functions defined in the sample.
```